code/KubernetesWrapper/KubernetesWrapper/Program.cs

```diff
@@ -0,0 +1,311 @@
+﻿using k8s;
+using k8s.Autorest;
+using k8s.Models;
+using System.Net;
+
+namespace KubernetesWrapper
+{
+    internal class PodList
+    {
+        private static async Task Main(string[] args)
+        {
+            var (jobName, containerName, imageName, namespaceName, ttlSecondsAfterFinished, command, arguments) = Util.ProcessArgs(args);
+            Console.WriteLine("Information about the job:");
+            Console.WriteLine($"Job Name: {jobName}");
+            Console.WriteLine($"Container Name: {containerName}");
+            Console.WriteLine($"Image Name: {imageName}");
+            Console.WriteLine($"Namespace Name: {namespaceName}");
+            Console.WriteLine("---------");
+
+            Console.WriteLine("Command: ");
+            Console.WriteLine($"length: {command.Count}");
+            foreach (var item in command)
+            {
+                Console.WriteLine(item);
+            }
+            Console.WriteLine("---------");
+
+            Console.WriteLine("Arguments: ");
+            Console.WriteLine($"length: {arguments.Count}");
+            foreach (var item in arguments)
+            {
+                Console.WriteLine(item);
+            }
+            Console.WriteLine("---------");
+
+            string? homeDirectory = Environment.GetEnvironmentVariable("HOME");
+            string kubeConfigPath = $"{homeDirectory}/.kube/config";
+            Console.WriteLine($"Home directory: {homeDirectory}");
+            Console.WriteLine($"Kube config path: {kubeConfigPath}");
+
+            var config = KubernetesClientConfiguration.BuildConfigFromConfigFile(kubeConfigPath);
+            IKubernetes client = new Kubernetes(config);
+
+            var nodes = Environment.GetEnvironmentVariable("CCP_NODES");
+            var nodeList = Util.GetNodeList(nodes);
+
+            if (nodeList.Count == 0)
+            {
+                Console.WriteLine("Node list is empty. Exiting...");
+                return;
+            }
+
+            Console.WriteLine("node list: ");
+            foreach (var item in nodeList)
+            {
+                Console.WriteLine(item);
+            }
+            Console.WriteLine("---------");
+
+            CancellationTokenSource source = new();
+            CancellationToken token = source.Token;
+
+            Console.CancelKeyPress += async (sender, e) =>
+            {
+                e.Cancel = true; // Prevent the process from terminating immediately
+                Console.WriteLine("interrupt!!");
+
+                try
+                {
+                    var job = await client.BatchV1.ReadNamespacedJobAsync(jobName, namespaceName);
```

Comment:
Would adding `.ConfigureAwait(false)` here and for other async calls help prevent potential deadlocks?

Reply:
There will be deadlock for UI application or ASP.Net application by default. I added `.ConfigureAwait(false)`, though it's recommended to do so for the library code. 
For reference: https://blog.stephencleary.com/2023/11/configureawait-in-net-8.html
"use ConfigureAwait(false) in library code and don’t use it in application code"